Property path

orneryd opened the pull request:
Opening the pull request to discuss the proposed property path changes.
1.  Adding a function to utils called "unwrapPropertyPath." this function searches an object for a path provided by "field" of a given column definition. This will enable users to specify an object hierarchy to the definition described in a standard object notation "object.property.path". Example:

columnDefs = ko.observable([
        { field: 'productInfo.Icon', displayName: 'Symbol', width: 50,},
        { field: 'productInfo.name.', displayName: 'Name', width: 110 },
        { field: 'productInfo.price', displayName: 'Price', width: 220 }]);
2. refactoring utils to clean up the global variable "utils". half of the code was accessing utils through kg.utils, and the other half was using utils as a globally assigned var.

ericmbarnard committed:
first commits on re-arch

orneryd committed:
refactoring utils and adding nested propertyPath functionality

usage is in the columnDef in the 'field' you put in the target nested
property:

{ field: 'Action.name.val', displayName: 'Action', width: 110 },

orneryd committed:
Pulling master into propertyPath

to fix up the auto-merge

orneryd committed:
fixing faulty merge

ericmbarnard commented on src/utils.js:
```diff
@@ -18,73 +18,87 @@
-                strMatch = "{" + intMatch.toString() + "}";
+        for (var j = 1; j < propPath.length; j++){
+            // entity could be observable or not...
+            if (ko.isObservable(tempProp)){
```
unwrap does this internally i believe, so you don't really need this if/check ... just straight up call `unwrapObservable`

orneryd committed:
fixing issue #44

specify a sorting algorithm that takes two variables and compares them,
returning 1, 0, or -1

orneryd committed:
removing redundant check and updating unwrap method to take the entity and not the row object.

That way the utility is more useful outside of this use case.

orneryd commented:
I removed the redundant check and made the utility more useful by taking an entity instead of the row.

orneryd commented:
aaaand......unit tests all pass
